client/ayon_houdini/api/pipeline.py

```diff
@@ -86,6 +86,11 @@ def install(self):
         #       opening with last workfile.
         _set_context_settings()
 
+        # Manually call on_new callback as it doesn't get called when AYON
+        # launches for the first time on a context, only when going to
+        # File -> New
+        on_new()
```

Comment:
Maybe, we can confirm whether this runs before or after the file open if "open last workfile" is triggering (like when launching Houdini with a filepath argument).

If this triggers before, it may be hard to detect here. If it's after, we may get by with using:
```python
if hou.hipFile.isNewFile():
    on_new()
```

I'm still not convinced that's the perfect approach - but at least you could test the different scenarios.
Also, I wonder if - because we really only want this logic on initialization of the app - whether it's worth it for this to make a separate function e.g. `on_init()` in which we put the logic that we want on init, but not to trigger on new each time (if it's in the `on_new` callback it's basically redundant resources triggering on each CTRL+N press?)

Reply:
it should also be running on new and not just on initialization of the app as you want the template scene to be used every time a new scene is created and not just on startup

Reply:
I think we should be using a combination of 123.py and the current `on_new` callback that currently only happens when creating a new scene after init

Reply:
Feel free to propose something that matches the constraints of when we want this to run or not and then we'll test it. I vaguely recall there were issues with `123.py` not running in certain versions of the software - which would be a massive hurdle.

As stated here: https://www.sidefx.com/docs/houdini/hom/locations.html the `123.py` ONLY runs for Houdini FX. Then you have `houdinicore.py` which ONLY runs for houdini core, but there's no equivalent for Houdini Apprentice if I recall :( So preferably we avoid using `123.py` or similar and somehow detect the right moment to trigger it on the app init.

> it should also be running on new and not just on initialization of the app as you want the template scene to be used every time a new scene is created and not just on startup

Valid point, makes sense.
I'm still hoping being able to use:
```python
if hou.hipFile.isNewFile():
    on_new()
```
Gets us somewhere if we can get it to run at the right time on launch. (and make sure it doesn't do anything in Deadline hehe)

Reply:
after all the conversations we had it seems like this is no longer a concern, am I right?